connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/MLUtils.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.planner
+
+import org.apache.spark.ml.param.{Param, Params}
+
+object SparkConnectUtils {
+
+  private def castParamValue(paramType: Class[_], paramValue: Object): Object = {
+    if (paramType == classOf[Int]) {
```

Comment:
Probably the answer is no, but can't this be done with `match`?

Reply:
Unfortunately, we can't .